jdaviz/configs/default/plugins/virtual_observatory/vo_plugin.py

```diff
@@ -0,0 +1,419 @@
+                        )
+                registry_args.append(
+                    registry.Spatial(
+                        (coord, (self.radius_val * u.Unit(self.radius_unit.selected))),
+                        intersect="overlaps",
+                    )
+                )
+            self._full_registry_results = registry.search(*registry_args)
+            self.resources = list(self._full_registry_results.getcolumn("short_name"))
+        except DALFormatError as e:
+            if type(e.cause) is RequestConnectionError:
+                self.hub.broadcast(
+                    SnackbarMessage(
+                        f"Can't connect to VO registry. Check your internet connection: {e}",
+                        sender=self,
+                        color="error",
+                    )
+                )
+            else:
+                raise e
+        except Exception as e:
+            self.hub.broadcast(
+                SnackbarMessage(
+                    f"An error occured querying the VO Registry: {e}",
+                    sender=self,
+                    color="error",
+                )
+            )
+            raise
+        finally:
+            self.resources_loading = False  # Stop loading bar
+
+    def vue_query_resource(self, _=None):
+        """
+        Once a specific VO resource is selected, query it with the user-specified source target.
+        User input for source is first attempted to be parsed as a SkyCoord coordinate. If not,
+        then attempts to parse as a target name.
+        """
+        # Reset Table
+        self.table.items = []
+        self._populate_url_only = False
+        self.table.headers_visible = ["Title", "Instrument", "DateObs"]
+
+        self.results_loading = True  # Start loading spinner
+        try:
+            # Query SIA service
+            # Service is indexed via short name (resource_selected), which is the suggested way
+            # according to PyVO docs. Though disclaimer that collisions COULD occur. If so,
+            # consider indexing on the full IVOID, which is guaranteed unique.
+            sia_service = self._full_registry_results[
+                self.resource_selected
+            ].get_service(service_type="sia")
+            try:
+                # First parse user-provided source as direct coordinates
+                coord = SkyCoord(
+                    self.source, unit=u.deg, frame=self.coordframe_selected
+                )
+            except Exception:
+                try:
+                    # If that didn't work, try parsing it as an object name
+                    coord = SkyCoord.from_name(
+                        self.source, frame=self.coordframe_selected
+                    )
+                except Exception:
+                    raise LookupError(
+                        f"Unable to resolve source coordinates: {self.source}"
+                    )
```

Comment:
same as above - are there specific exceptions we can catch?

Reply:
Ack; forgot to add a snackbar message here! Added in fbfaa1ddb70afb3f1f21a3864274e200d0e2119a

Otherwise, same response as https://github.com/spacetelescope/jdaviz/pull/2872/files#r1815657235